### PR TITLE
add some logs

### PR DESCRIPTION
**Pull request by zemtsov**
- Changed ACL methods signature. Now they return `[]byte, error` or `error` instead of `peer.Response`. 
- Updated error messages in all methods.
- Added some logs to `Invoke`.

### CODE REVIEW

File: cc/account_info.go [hunk 1]
```diff
@@ -23,83 +22,92 @@ import (
...
+		return nil, fmt.Errorf("failed to get account info composite key: %w", err)
 	}
+
 	if err = stub.PutState(cKey, infoMsg); err != nil {
-		return shim.Error(err.Error())
+		return nil, fmt.Errorf("failed to put account info message into state: %w", err)
 	}
 
-	return shim.Success(nil)
+	return nil, nil
```
Comment (andreevym) [inline cc/account_info.go:90]: nil nil

File: cc/account_info.go [hunk 2]
```diff
@@ -23,83 +22,92 @@ import (
 // arg[1] - KYC hash
 // arg[2] - is address gray listed? ("true" or "false")
 // arg[3] - is address black listed? ("true" or "false")
-func (c *ACL) SetAccountInfo(stub shim.ChaincodeStubInterface, args []string) peer.Response {
+func (c *ACL) SetAccountInfo(stub shim.ChaincodeStubInterface, args []string) ([]byte, error) {
```
Comment (andreevym) [inline cc/account_info.go:25]: здесь нужно возвращать только error

File: cc/additional_keys.go [hunk 3]
```diff
@@ -158,15 +158,15 @@ func (c *ACL) AddAdditionalKey(
...
 	}
 
 	// Saves a link to the parent address.
 	if err = stub.PutState(additionalKeyParentComposite, []byte(userAddress)); err != nil {
-		return errF("put state (parent link address) for %s: %s", userAddress, err)
+		return nil, fmt.Errorf("put state (parent link address) for %s: %w", userAddress, err)
 	}
 
-	return shim.Success(nil)
+	return nil, nil
```
Comment (andreevym) [inline cc/additional_keys.go:169]: nil, nil

File: cc/multisign.go [hunk 4]
```diff
@@ -21,21 +21,21 @@ import (
...
+		return nil, fmt.Errorf("failed parsing arguments: %w", err)
 	}
 
 	if err = addMultisig(stub, request); err != nil {
-		return shim.Error(err.Error())
+		return nil, fmt.Errorf("failed adding multisig: %w", err)
 	}
 
-	return shim.Success(nil)
+	return nil, nil
```
Comment (andreevym) [inline cc/multisign.go:38]: nil, nil

File: cc/public_key.go [hunk 5]
```diff
@@ -22,7 +22,7 @@ type PublicKey struct {
 func publicKeyFromBase58String(base58Encoded string) (PublicKey, error) {
 	bytes, err := helpers.DecodeBase58PublicKey(base58Encoded)
 	if err != nil {
-		return PublicKey{}, fmt.Errorf("failed decoding public key: %w", err)
+		return PublicKey{}, err
```
Comment (andreevym) [inline cc/public_key.go:25]: missed wrap err

File: cc/multisign.go [hunk 6]
```diff
@@ -187,42 +188,42 @@ func (c *ACL) ChangeMultisigPublicKey(stub shim.ChaincodeStubInterface, args []s
...
 	}
 
 	// set new address -> key mapping
 	if err = stub.PutState(addrToPkCompositeKey, []byte(hashedHexKeys)); err != nil {
-		return shim.Error(err.Error())
+		return nil, fmt.Errorf("failed putting public keys to state: %w", err)
 	}
 
-	return shim.Success(nil)
+	return nil, nil
```
Comment (andreevym) [inline cc/multisign.go:228]: nil, nil

File: cc/multisign.go [hunk 7]
```diff
@@ -21,21 +21,21 @@ import (
 // arg[1] nonce
 // args[2:] are the public keys and signatures hex of all participants in the multi-wallet
 // and signatures confirming the agreement of all participants with the signature policy
-func (c *ACL) AddMultisig(stub shim.ChaincodeStubInterface, args []string) peer.Response {
+func (c *ACL) AddMultisig(stub shim.ChaincodeStubInterface, args []string) ([]byte, error) {
```
Comment (andreevym) [inline cc/multisign.go:24]: return only error

File: tests/unit/black_list_test.go [hunk 8]
```diff
@@ -82,7 +82,7 @@ func TestBlackListWrongParameterList(t *testing.T) {
 		respStatus:  int32(shim.ERROR),
 	}
 
-	errorMsg := "%s is not valid list type, accepted 'black' or 'gray' only"
+	errorMsg := "kek is not valid list type, accepted 'black' or 'gray' only"
```
Comment (andreevym) [inline tests/unit/black_list_test.go:85]: kek?

File: cc/access_matrix.go [hunk 9]
```diff
@@ -150,15 +150,15 @@ func (c *ACL) AddRights(stub shim.ChaincodeStubInterface, args []string) peer.Re
...
 		}
 		err = stub.PutState(accountCompositeKey, rawAccountRights)
 		if err != nil {
-			return shim.Error(err.Error())
+			return nil, fmt.Errorf("failed putting account rights to state: %w", err)
 		}
 	}
 
-	return shim.Success(nil)
+	return nil, nil
```
Comment (pfi79) [inline cc/access_matrix.go:161]: Return nil, nil is bad

File: cc/access_matrix.go [hunk 10]
```diff
@@ -255,17 +255,17 @@ func (c *ACL) RemoveRights(stub shim.ChaincodeStubInterface, args []string) peer
...
 			if err != nil {
-				return shim.Error(err.Error())
+				return nil, fmt.Errorf("failed putting account rights to state: %w", err)
 			}
 			break
 		}
 	}
 
-	return shim.Success(nil)
+	return nil, nil
```
Comment (pfi79) [inline cc/access_matrix.go:268]: Return nil, nil is bad

File: cc/account_info.go (hunk 1 above)
Comment (pfi79) [inline cc/account_info.go:90]: Return nil, nil is bad

File: cc/acl.go [hunk 11]
```diff
@@ -39,23 +43,23 @@ type AddrsWithPagination struct {
...
+		return nil, fmt.Errorf("failed parsing arguments: %w", err)
 	}
 
 	if err = addUser(stub, request); err != nil {
-		return shim.Error(err.Error())
+		return nil, fmt.Errorf("failed adding new user: %w", err)
 	}
 
-	return shim.Success(nil)
+	return nil, nil
```
Comment (pfi79) [inline cc/acl.go:62]: Return nil, nil is bad

File: cc/acl.go [hunk 12]
```diff
@@ -65,105 +69,105 @@ func (c *ACL) AddUser(stub shim.ChaincodeStubInterface, args []string) peer.Resp
...
```
Comment (pfi79) [inline cc/acl.go:88]: Return nil, nil is bad

File: cc/acl.go [hunk 13]
```diff
@@ -251,107 +255,121 @@ func (c *ACL) updateSignedAddress(
...
+		return nil, fmt.Errorf("failed marshalling account info: %w", err)
 	}
 
 	if err = stub.PutState(cKey, newAccInfo); err != nil {
-		return shim.Error(err.Error())
+		return nil, fmt.Errorf("failed storing account info: %w", err)
 	}
 
-	return shim.Success(nil)
+	return nil, nil
```
Comment (pfi79) [inline cc/acl.go:335]: Return nil, nil is bad

File: cc/acl.go [hunk 14]
```diff
@@ -392,166 +411,166 @@ func (c *ACL) GetAddresses(stub shim.ChaincodeStubInterface, args []string) peer
...
 	}
 
 	// set new address -> key mapping
 	if err = stub.PutState(addrToPkCompositeKey, []byte(newKey)); err != nil {
-		return shim.Error(err.Error())
+		return nil, fmt.Errorf("failed putting new public key to state: %w", err)
 	}
 
-	return shim.Success(nil)
+	return nil, nil
```
Comment (pfi79) [inline cc/acl.go:573]: Return nil, nil is bad

File: cc/acl.go [hunk 15]
```diff
@@ -561,147 +580,150 @@ func (c *ACL) ChangePublicKeyWithBase58Signature(stub shim.ChaincodeStubInterfac
...
```
Comment (pfi79) [inline cc/acl.go:726]: Return nil, nil is bad

File: cc/additional_keys.go [hunk 16]
```diff
@@ -158,15 +157,15 @@ func (c *ACL) AddAdditionalKey(
...
```
Comment (pfi79) [inline cc/additional_keys.go:168]: Return nil, nil is bad

File: cc/additional_keys.go [hunk 17]
```diff
@@ -290,70 +289,70 @@ func (c *ACL) RemoveAdditionalKey(stub shim.ChaincodeStubInterface, args []strin
...
 	}
 
 	// Removing the link to the parent address.
 	if err = stub.DelState(additionalKeyParentComposite); err != nil {
-		return errF("delete state (parent link address) for %s: %s", userAddress, err)
+		return nil, fmt.Errorf("delete state (parent link address) for %s: %w", userAddress, err)
 	}
 
-	return shim.Success(nil)
+	return nil, nil
```
Comment (pfi79) [inline cc/additional_keys.go:300]: Return nil, nil is bad

File: cc/additional_keys.go [hunk 18]
```diff
@@ -290,70 +289,70 @@ func (c *ACL) RemoveAdditionalKey(stub shim.ChaincodeStubInterface, args []strin
...
+) ([]byte, error) {
 	const (
 		argsLen            = 1
 		multisignSeparator = "/"
 	)
 
 	// Checking that the argument is the only one needed for the extra key case.
 	if len(args) != argsLen {
-		return resp, false
+		return nil, nil
```
Comment (pfi79) [inline cc/additional_keys.go:314]: Return nil, nil is bad

File: cc/additional_keys.go [hunk 19]
```diff
@@ -290,70 +289,70 @@ func (c *ACL) RemoveAdditionalKey(stub shim.ChaincodeStubInterface, args []strin
...
 	}
 
 	// Query Parameters.
 	publicKey := args[0]
 
 	// Check if the argument is a multisignature.
 	if strings.Count(publicKey, multisignSeparator) > 0 {
-		return resp, false
+		return nil, nil
```
Comment (pfi79) [inline cc/additional_keys.go:322]: Return nil, nil is bad

File: cc/additional_keys.go [hunk 20]
```diff
@@ -290,70 +289,70 @@ func (c *ACL) RemoveAdditionalKey(stub shim.ChaincodeStubInterface, args []strin
...
 	parentAddress, _, err := c.retrieveParentAddress(stub, publicKey)
 	if err != nil {
-		return errF("get parent address for %s: %s", publicKey, err), true
+		return nil, fmt.Errorf("get parent address for %s: %w", publicKey, err)
 	}
 
 	// If no parent is found, the key is normal and control is passed to the higher handler.
 	if parentAddress == "" {
-		return resp, false
+		return nil, nil
```
Comment (pfi79) [inline cc/additional_keys.go:333]: Return nil, nil is bad

File: cc/chaincode.go [hunk 21]
```diff
@@ -8,27 +8,38 @@ import (
...
 	"runtime/debug"
+	"time"
 
 	"github.com/anoideaopen/acl/helpers"
 	"github.com/anoideaopen/acl/internal/config"
 	"github.com/anoideaopen/acl/proto"
+	"github.com/anoideaopen/foundation/core/logger"
 	"github.com/hyperledger/fabric-chaincode-go/shim"
 	"github.com/hyperledger/fabric-protos-go/peer"
+	"github.com/op/go-logging"
```
[the remark this answers is not in the document]
Comment (zemtsov) [reply cc/chaincode.go:19]: this packed is used in foundation

[the remark this answers is not in the document]
Comment (pfi79) [reply cc/chaincode.go:19]: very strange

[the remark this answers is not in the document]
Comment (zemtsov) [reply cc/chaincode.go:19]: I invoke function Logger from foundation and it returns a type from this package.

**Review by pfi79 — APPROVED** [review]

**Pull request merged**